jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2579,6 +2580,90 @@ ecma_builtin_array_prototype_object_copy_within (const ecma_value_t args[], /**<
 
   return ecma_copy_value (ecma_make_object_value (obj_p));
 } /* ecma_builtin_array_prototype_object_copy_within */
+
+/**
+ * The Array.prototype object's 'includes' routine
+ *
+ * See also:
+ *          ECMA-262 v11, 22.1.3.13
+ *
+ * @return ECMA_VALUE_ERROR -if the operation fails
+ *         ECMA_VALUE_{TRUE/FALSE} - depends on whether the search element is in the array or not
+ */
+static ecma_value_t
+ecma_builtin_array_prototype_includes (const ecma_value_t args[], /**< arguments list */
+                                       ecma_length_t args_number, /**< number of arguments */
+                                       ecma_object_t *obj_p, /**< array object */
+                                       uint32_t len) /**< array object's length */
+{
+  /* 3. */
+  if (len == 0)
+  {
+    return ECMA_VALUE_FALSE;
+  }
+
+  uint32_t from_index = 0;
+
+  /* 4-7. */
+  if (args_number > 1)
+  {
+    if (ECMA_IS_VALUE_ERROR (ecma_builtin_helper_array_index_normalize (args[1], len, &from_index)))
+    {
+      return ECMA_VALUE_ERROR;
+    }
+  }
+
+  /* Fast array path */
+  if (ecma_op_object_is_fast_array (obj_p))
+  {
+    ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+
+    if (ext_obj_p->u.array.u.hole_count < ECMA_FAST_ARRAY_HOLE_ONE)
+    {
+      if (obj_p->u1.property_list_cp != JMEM_CP_NULL)
+      {
+        len = JERRY_MIN (ext_obj_p->u.array.length, len);
```

Comment:
In most of the cases these are the same, but if `argv[1]` is an object which `valueof` property is a callback which changes the length of the array this line is significant. please add testcase for it as well.